web3.js/src/vote-account.ts

```diff
@@ -183,26 +216,21 @@ function parsePriorVoters({
   authorizedPubkey,
   epochOfLastAuthorizedSwitch,
   targetEpoch,
-}: PriorVoter) {
+}: PriorVoterRaw): PriorVoter {
   return {
     authorizedPubkey: new PublicKey(authorizedPubkey),
     epochOfLastAuthorizedSwitch,
     targetEpoch,
   };
 }
 
-function getPriorVoters({
-  buf,
-  idx,
-  isEmpty,
-}: {
-  buf: PriorVoter[];
-  idx: number;
-  isEmpty: boolean;
-}): PriorVoter[] {
+function getPriorVoters({buf, idx, isEmpty}: PriorVoters): PriorVoter[] {
   if (isEmpty) {
     return [];
   }
 
-  return [...buf.slice(idx + 1).map(parsePriorVoters), ...buf.slice(0, idx)];
```

Comment:
I'm not super sure how this code was getting away with _not_ parsing the second chunk of prior voters. This might have been a bug that was previously hidden by the incorrect typing of the input arguments to this function.